config/secretsmanager/config.go

```diff
@@ -13,5 +22,193 @@ func Configure(p *config.Provider) {
+						removeReplicaFromDiffViaHash(region[dReplica.KMSKeyID])
+						createdHash := hashFunc(map[string]interface{}{
+							"kms_key_id": dReplica.KMSKeyID,
+							"region":     dReplica.Region,
+						})
+						removeReplicaFromDiffViaHash(createdHash)
+						delete(region, dReplica.KMSKeyID)
+					}
+				}
+			}
+
+			// now try to match the region-only desired KMS Key IDs with the ones in the left in current state (after filtering out the explicit matches above).
+			for region, unmatchedDesiredKMSKeyCount := range desiredReplicasWithRegionOnly {
+				KMSKeysOfRegionInCurrentState, ok := regionsInCurrentState[region]
+				if !ok {
+					// this is a Replica with brand-new region, no action needed. it already shows up on diff
+					continue
+				}
+
+				switch {
+				case len(KMSKeysOfRegionInCurrentState) > unmatchedDesiredKMSKeyCount:
+					// for the particular region, we have more KMS Key IDs present in the current state than we desire, e.g.
+					// current state for region1 = { region1_kmsKeyA, region1_kmsKeyB, region1_kmsKeyC }
+					// desired state for region1 = { region1_kmsKeyANY }
+					// due to set difference implementation in TF, this will show up as 3 deletions, 1 creation in DIFF
+					// instead, in this case, we want to have 2 deletions, 0 creation DIFF
+					// thus, remove all (unmatchedDesiredKMSKeyCount=1) creation diffs and remove (unmatchedDesiredKMSKeyCount=1) deletion diff
+					// choose Replica to delete randomly
+					i := 0
+					for _, hash := range KMSKeysOfRegionInCurrentState {
+						if i >= unmatchedDesiredKMSKeyCount {
+							break
+						}
+						removeReplicaFromDiffViaHash(hash)
+						i++
+					}
+					creationHash := hashFunc(map[string]interface{}{
+						"kms_key_id": "",
+						"region":     region,
+					})
+					removeReplicaFromDiffViaHash(creationHash)
+				case len(KMSKeysOfRegionInCurrentState) < unmatchedDesiredKMSKeyCount:
+					// this might not be possible at all, due to Replica hash function
+					for _, hash := range KMSKeysOfRegionInCurrentState {
+						removeReplicaFromDiffViaHash(hash)
+					}
+				default:
+					// for the particular region, we have matching number of KMS Key IDs to desired, i.e. there should be no diff for these
+					// example
+					// current state for region2 KMS Key IDs = { region2_kmsKeyX}
+					// desired state for region2 KMS Key IDs = { region2_kmsKeyANY }
+					// due to set difference implementation in TF, this will show up as 1 deletion, 1 creation in DIFF
+					// instead, in this case, we want to have no diff
+					// thus, remove all creation diffs and remove all deletion diffs
+					// choose Replica to delete randomly
+					for _, hash := range KMSKeysOfRegionInCurrentState {
+						removeReplicaFromDiffViaHash(hash)
+					}
+					creationHash := hashFunc(map[string]interface{}{
+						"kms_key_id": "",
+						"region":     region,
+					})
+					removeReplicaFromDiffViaHash(creationHash)
```

Comment:
theoratically, in the generic logic, this should remove all creation diffs, however practically this is never > 1. realized that we anyways cannot specify two replicas in the same region without kms ids specified in the config due to they end up as same hash and a single element. So, something like below is not possible to specify.
```yaml
apiVersion: secretsmanager.aws.upbound.io/v1beta1
kind: Secret
metadata:
  name: my-secret
spec:
  deletionPolicy: Delete
  forProvider:
    name: my-secret
    region: us-east-1
    replica:
    - region: us-east-2
    - region: us-east-2
```
Anyways, this is not valid from the cloud API perspective, as you can have a single replica per region.